datalad/interface/open.py

```diff
@@ -0,0 +1,177 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""A context manager to assist with annexed files access (i.e. read/write)"""
+
+__docformat__ = 'restructuredtext'
+
+import os
+import logging
+
+from datalad.distribution.dataset import resolve_path
+from datalad.distribution.dataset import datasetmethod
+from datalad.utils import assure_list
+
+
+lgr = logging.getLogger('datalad.interface.open')
+
+_builtin_open = open
```

Comment:
I wonder if we should use `io.open` and make this open support the full set of parameters.  One downside of that is that it'd be slower for Python 2, which I would obviously weigh different than you since you still primarily use py2.

Reply:
I thought that `io.open` is considered an "internal" implementation and `open` generally recommended instead... I could be wrong - will check later

Reply:
> I thought that io.open is considered an "internal" implementation and open generally recommended instead...

I'm not aware of that and didn't find anything that reflects the in the Python docs.  My understanding is that `io.open` is useful on py2 for py3-compatibility.  On py3, `io.open` is simply an alias for `open`.

Reply:
That is my understanding too.

Reply:
ok, let's proxy `io.open` then